lib/events/session_writer.go

```diff
@@ -375,11 +370,33 @@ func (a *AuditWriter) EmitAuditEvent(ctx context.Context, event apievents.AuditE
 	}
 }
 
+func checkEventFields(event apievents.AuditEvent) error {
+	if err := checkBasicEventFields(event); err != nil {
+		return trace.Wrap(err)
+	}
+	if event.GetTime().IsZero() {
+		return trace.BadParameter("Time is unset, call SetupEvent before recording events")
```

Comment:
This is not a good message for a user - it doesn't tell them what's gone wrong or how to fix it.

Reply:
I agree, though my goal for these errors is to let devs know when they forgot to call `SetupEvent` first. We really need some kind of debug-panic log or something, so we can panic and let developers know when they're not using something correctly without panicking when run in production.